### PR TITLE
SPARK-99531-[Web Client] Add blocking qualifiers in Clients when accessing files for download

### PR DESCRIPTION
**Pull request by meeshaT**
# Pull Request Template

## Description

For POST /download/endpoints the parameter with **allow=unchecked** should be put into the request body as an array. This will allow unscanned files to be downloaded without a warning, but files currently being scanned or already determined to be infected will evoke 423 or 410 responses.

If the client is using the Files APIs to download thumbnails or preview pages, then need to add "allow=unchecked, evaluating" in the new field in the POST body accordingly.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

**Test Configuration**:
* SDK Version 1.80.29
* Node/Browser Version 1.80.29
* NPM Version 6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js [hunk 1]
```diff
@@ -65,14 +67,14 @@ const Encryption = WebexPlugin.extend({
...
-  download(scr) {
+  download(scr, optional) {
     /* istanbul ignore if */
     if (!scr.loc) {
       return Promise.reject(new Error('`scr.loc` is required'));
     }
 
     const shunt = new EventEmitter();
-    const promise = this._fetchDownloadUrl(scr)
+    const promise = this._fetchDownloadUrl(scr, optional)
```
Comment (andrewholsted) [inline packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js]: The general naming convention for this param would be `options`. `optional` would be more of a boolean value and this looks like it takes an object. Also, adding some code comments or js-docs would greatly explain why/what is happening

File: package-lock.json [hunk 2]
```diff
@@ -6924,8 +6924,7 @@
         },
         "ansi-regex": {
           "version": "2.1.1",
-          "bundled": true,
-          "optional": true
+          "bundled": true
```
Comment (InteractiveTimmy) [inline package-lock.json]: Not sure if this change to the package-lock.json file is needed, as I don't see any additional dependencies or upstream changes from the scope of this PR. If these changes are required, you can ignore this comment.

File: packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js [hunk 3]
```diff
@@ -2,13 +2,15 @@
...
 
+import {EventEmitter} from 'events';
+
 import {WebexPlugin} from '@webex/webex-core';
 import {proxyEvents, tap, transferEvents} from '@webex/common';
-import {EventEmitter} from 'events';
 import jose from 'node-jose';
 import SCR from 'node-scr';
-import ensureBuffer from './ensure-buffer';
+import isUndefined from 'lodash/isUndefined';
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js]: Not sure if this is needed, there are ways to determine if an object or value is undefined without using lodash, such as `(!value)` or `(typeof value === 'undefined')`, as well as many other options. Without JSDoc, it cannot be determined which approach is best.

File: packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js [hunk 4]
```diff
@@ -65,14 +67,14 @@ const Encryption = WebexPlugin.extend({
         .then((result) => result.plaintext.toString()));
   },
 
-  download(scr) {
+  download(scr, optional) {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js]: As @andrewholsted mentioned, JSDoc comments would help contributors understand what this method is doing, and what it requires.

File: packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js [hunk 5]
```diff
@@ -93,24 +95,33 @@ const Encryption = WebexPlugin.extend({
     return promise;
   },
 
-  _fetchDownloadUrl(scr) {
+  _fetchDownloadUrl(scr, optional) {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js]: Please add JSDoc comments for this method as well.

File: packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js [hunk 6]
```diff
@@ -93,24 +95,33 @@ const Encryption = WebexPlugin.extend({
...
     this.logger.info('encryption: retrieving download url for encrypted file');
+    let inputBody;
 
     if (process.env.NODE_ENV !== 'production' && scr.loc.includes('localhost')) {
       this.logger.info('encryption: bypassing webex files because this looks to be a test file on localhost');
 
       return Promise.resolve(scr.loc);
     }
 
+    if (isUndefined(optional)) {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js]: See my comments on `lodash` and apply them here if possible.

File: packages/node_modules/@webex/internal-plugin-encryption/test/integration/spec/encryption.js [hunk 7]
```diff
@@ -173,6 +173,38 @@ describe('Encryption', function () {
           .on('progress', spy))
         .then(() => assert.called(spy));
     });
+
+    it('checks body of the API call /downloads/endpoints', () => {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-encryption/test/integration/spec/encryption.js]: If you could, write a test or two for `_fetchDownloadUrl()` to help us improve coverage for that method in the event that another contributor makes changes to this section of code.

**Comment by arpydi** [comment]
Hey @andrewholsted @InteractiveTimmy.

As @meeshaT is on a long leave, I made the required changes and raised a new PR #1401 here.
Please don't merge this PR.
Please check the other one here: https://github.com/webex/webex-js-sdk/pull/1401

**Pull request closed without merging**